core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -30,3 +30,12 @@ class SparkException(message: String, cause: Throwable)
  */
 private[spark] class SparkDriverExecutionException(cause: Throwable)
   extends SparkException("Execution error", cause)
+
+/**
+ * Exception indicating an error internal to Spark, not caused by user error
+ */
+class SparkInternalStateException(message: String, cause: Throwable)
```

Comment:
This name sounds weird, but I guess `SparkIllegalInternalStateException` would be a little long. Maybe `SparkIllegalStateException`?